lib/game.rb

```diff
@@ -0,0 +1,100 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2], # Top row
+    [3,4,5], # Middle row
+    [6,7,8], # Bottom row
+    [0,3,6], # Left col
+    [1,4,7], # Middle col
+    [2,5,8], # Right col
+    [0,4,8], # Left top to right bottom
+    [2,4,6], # Right top to left bottom
+  ]
+
+  def initialize(player1 = Players::Human.new("X"), player2=Players::Human.new("O"), board = Board.new)
+    @player_1 = player1
+    @player_2 = player2
+    @board = board
+  end
+
+  def current_player
+    # @board should have a turn_count method
+
+    if @board.turn_count % 2 == 0
+      @player_1
+    else
+      @player_2
+    end
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def winner
+    # if won?
+    #   #win = won? # won? returns the winning array
+    #   @board.cells[won?[0]] # take an index from the winning array to show the winners token(X/O)
+    # elsif !over? || !@board.full?
+    #   nil
+    # end
+    
+    # if !won? || draw?
+    #   binding.pry
+    # elsif over? && won?
+    #   @board.cells[won?[0]]
+    # end
+
+    if won?
+      @winner = @board.cells[won?.first]
+    end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.